app/src/main/java/nl/tudelft/b_b_w/controller/BlockController.java

```diff
@@ -59,14 +75,45 @@ public BlockController(Context _context) {
         return getBlocks(owner);
     }
 
+
     /**
-     * Get all blocks that are not revoked
+     * Add a block to the database with checking if the (owner,pubkey) pair
+     * is already added to the database
+     *
+     * @param block Block you want to add
+     */
+    public void addBlock(Block block) {
+
+        if (blockExists(block.getOwner(), block.getPublicKey(), block.isRevoked()))
```

Comment:
Yes, this is why the block.isRevoked() also gets passed through to the function; if there is a normal block but not a revoke, the blockExists function still returns false.